### PR TITLE
Fix issue when there is only one note-attribute

### PR DESCRIPTION
**Pull request by Cito**
In this case, xml-flow collapses the note-attribute sub-tag which will cause the corresponding note-attribute to be missing.

Fixed this edge-case and added a test spec "note with only source url". Without the change in `yarle.ts`, the test will fail.

Also added source-url to the default template, because I think the source URL is always important when there is one.

### CODE REVIEW

File: src/yarle.ts [hunk 1]
```diff
@@ -31,34 +31,42 @@ export const parseStream = async (options: YarleOptions): Promise<void> => {
...
-    
+
     const logAndReject = (error: Error) => {
       console.log(`Could not convert ${options.enexSource}:\n${error.message}`);
       ++failed;
 
       return reject();
     };
-    if (!fs.existsSync(options.enexSource))
+    if (!fs.existsSync(options.enexSource)) {
```
Comment (akosbalasko) [inline src/yarle.ts:45]: Could you please push your tslint config? then we could work using the same settings. Thank you!

File: test/utils.spec.ts [hunk 2]
```diff
@@ -43,12 +43,12 @@ describe('SetFileDates', () => {
...
-            const referTime = moment();
-            assert.equal(atime, referTime.format());
-            assert.equal(mtime, referTime.format());
-
+            const atime = moment(fStat.atime);
+            const mtime = moment(fStat.mtime);
+            const referTimeLo = moment().subtract(3, 's');
+            const referTimeHi = moment().add(3, 's');
+            assert.ok(atime.isBetween(referTimeLo, referTimeHi));
+            assert.ok(mtime.isBetween(referTimeLo, referTimeHi));
```
Comment (akosbalasko) [inline test/utils.spec.ts:51]: good trick :)

**Review by akosbalasko — CHANGES_REQUESTED** [review]

File: src/yarle.ts (hunk 1 above)
Comment (Cito) [reply src/yarle.ts:45]: Hm, I think I'm using the one that is [in the project](https://github.com/akosbalasko/yarle/blob/master/tslint.json) already. This is the "curly" rules that says "if statements must be braced".

File: src/yarle.ts [hunk 3]
```diff
@@ -31,34 +31,42 @@ export const parseStream = async (options: YarleOptions): Promise<void> => {
...
       return logAndReject({name: 'NoSuchFileOrDirectory', message: 'source Enex file does not exists'});
-    
+    }
+
     const xml = flow(stream);
```
Comment (Cito) [inline src/yarle.ts:49]: Btw, you can pass some options to `flow()`, but I found now option that prevented the collapse of the note-attributes tag. Except maybe preserveMarkup, but that makes everything complicated again.

File: src/yarle.ts (hunk 1 above)
Comment (akosbalasko) [reply src/yarle.ts:45]: Hm. okay, I don't know then, I'll recheck my vscode's tslint component

**Review by akosbalasko — APPROVED** [review]
